sdk/program/src/rent.rs

```diff
@@ -68,12 +68,17 @@ impl Rent {
         if self.is_exempt(balance, data_len) {
             RentDue::Exempt
         } else {
-            let actual_data_len = data_len as u64 + ACCOUNT_STORAGE_OVERHEAD;
-            let lamports_per_year = self.lamports_per_byte_year * actual_data_len;
-            RentDue::Paying((lamports_per_year as f64 * years_elapsed) as u64)
+            self.paying_due(data_len, years_elapsed)
         }
     }
 
+    /// rent due for account that is known to be not exempt
+    pub fn paying_due(&self, data_len: usize, years_elapsed: f64) -> RentDue {
+        let actual_data_len = data_len as u64 + ACCOUNT_STORAGE_OVERHEAD;
+        let lamports_per_year = self.lamports_per_byte_year * actual_data_len;
+        RentDue::Paying((lamports_per_year as f64 * years_elapsed) as u64)
+    }
```

Comment:
A few thoughts:

1. Since this is in the SDK, I don't know what the process is/impacts are for add `pub` functions. Do you know if there are any concerns here?

2. What if this function instead calculates the rent amount (as `u64`) instead of a `RentDue::Paying`? So:
```suggestion
    /// rent due for account that is known to be not exempt
    pub fn due_amount(&self, data_len: usize, years_elapsed: f64) -> u64 {
        let actual_data_len = data_len as u64 + ACCOUNT_STORAGE_OVERHEAD;
        let lamports_per_year = self.lamports_per_byte_year * actual_data_len;
        (lamports_per_year as f64 * years_elapsed) as u64
    }
```

Then `fn due()` would change the `else` block to be:
```rust
} else {
    RentDue::Paying(self.due_amount(data_len, years_elapsed))
}
```

and similarly `RentCollector::get_rent_due()` would change to:
```rust
// we know this account is not exempt
RentDue::Paying(self.rent.due_amount(account.data().len(), years_elapsed))
```

This would make it such that the if and else blocks are matched in that they both explicitly state which `RentDue` variant they return.

Reply:
# 2 is done

Reply:
@ryoqun I do not have insight into adding a method to sdk. Do you?

Reply:
For #1, I asked [over in Discord](https://discord.com/channels/428295358100013066/476811830145318912/946895787521101904). Sounds like this is a-ok.